File: Common/BusinessObjects/DrivingTest.cs
﻿using System;
using System.Collections.Generic;
using Common.Entities;

namespace Common.BusinessObjects
{
    public class DrivingTest
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public User User { get; set; }

        public int RuleSectionId { get; set; }
        
        public RuleSection RuleSection { get; set; }

        public ICollection<RoadSituation> RoadSituations { get; set; }

        public DateTime AddedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public bool? Success { get; set; }

        public DrivingTest()
        {
            RoadSituations = new List<RoadSituation>();
        }
    }
}

File: AutoSchool/ClientApp/src/app/shared/entities/DrivingTest.ts
import { User } from './User';
import { TestRules } from './TestRules';
import { formatDate } from '@angular/common';
import { RuleSection } from './RuleSection';
import { RoadSituation } from './RoadSituation';

export class DrivingTest {
    constructor(
        public id?: number,
        public userId?: number,
        public ruleSectionName?: string,
        public roadSituations?: RoadSituation[],
        public addedAt?: Date,
        public updatedAt?: Date,
        public success?: boolean
    ) { }
}
